#! /bin/sh /usr/share/dpatch/dpatch-run
## 14_kfreebsd_build_fix.dpatch by Petr Salinger <Petr.Salinger@seznam.cz>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes FTBFS on GNU/kFreeBSD by creating new Q_OS_GLIBC.

@DPATCH@
diff -urNad qt4-x11-4.3.0~rc1~/src/corelib/global/qglobal.h qt4-x11-4.3.0~rc1/src/corelib/global/qglobal.h
--- qt4-x11-4.3.0~rc1~/src/corelib/global/qglobal.h	2007-05-06 17:54:22.000000000 +0200
+++ qt4-x11-4.3.0~rc1/src/corelib/global/qglobal.h	2007-05-18 10:48:20.000000000 +0200
@@ -110,6 +110,12 @@
 #  define Q_OS_RELIANT
 #elif defined(__linux__) || defined(__linux)
 #  define Q_OS_LINUX
+#  define Q_OS_GLIBC
+#elif defined(__GNU_HURD__) || defined(__GNU__)
+#  define Q_OS_HURD
+#  define Q_OS_GLIBC
+#elif defined(__GLIBC__)
+#  define Q_OS_GLIBC
 #elif defined(__FreeBSD__) || defined(__DragonFly__)
 #  define Q_OS_FREEBSD
 #  define Q_OS_BSD4
@@ -130,8 +136,6 @@
 #  define Q_OS_AIX
 #elif defined(__Lynx__)
 #  define Q_OS_LYNX
-#elif defined(__GNU__)
-#  define Q_OS_HURD
 #elif defined(__DGUX__)
 #  define Q_OS_DGUX
 #elif defined(__QNXNTO__)
diff -urNad qt4-x11-4.3.0~rc1~/src/corelib/plugin/qlibrary.cpp qt4-x11-4.3.0~rc1/src/corelib/plugin/qlibrary.cpp
--- qt4-x11-4.3.0~rc1~/src/corelib/plugin/qlibrary.cpp	2007-05-06 17:54:24.000000000 +0200
+++ qt4-x11-4.3.0~rc1/src/corelib/plugin/qlibrary.cpp	2007-05-18 10:50:38.000000000 +0200
@@ -278,11 +278,11 @@
 
 #if defined(Q_OS_UNIX) && !defined(Q_OS_MAC)
 
-#if defined(Q_OS_FREEBSD) || defined(Q_OS_LINUX)
+#if defined(Q_OS_FREEBSD) || defined(Q_OS_GLIBC)
 #  define USE_MMAP
 #  include <sys/types.h>
 #  include <sys/mman.h>
-#endif // Q_OS_FREEBSD || Q_OS_LINUX
+#endif // Q_OS_FREEBSD || Q_OS_GLIBC
 
 static long qt_find_pattern(const char *s, ulong s_len,
                              const char *pattern, ulong p_len)
diff -urNad qt4-x11-4.3.0~rc1~/src/corelib/tools/qlocale.cpp qt4-x11-4.3.0~rc1/src/corelib/tools/qlocale.cpp
--- qt4-x11-4.3.0~rc1~/src/corelib/tools/qlocale.cpp	2007-05-06 17:54:24.000000000 +0200
+++ qt4-x11-4.3.0~rc1/src/corelib/tools/qlocale.cpp	2007-05-18 10:51:21.000000000 +0200
@@ -48,7 +48,7 @@
 #include <qdebug.h>
 #include <time.h>
 
-#if defined(Q_OS_LINUX) && !defined(__UCLIBC__)
+#if defined(Q_OS_GLIBC) && !defined(__UCLIBC__)
 #    include <fenv.h>
 #endif
 
@@ -5906,7 +5906,7 @@
     _control87(MCW_EM, MCW_EM);
 #endif
 
-#if defined(Q_OS_LINUX) && !defined(__UCLIBC__)
+#if defined(Q_OS_GLIBC) && !defined(__UCLIBC__)
     fenv_t envp;
     feholdexcept(&envp);
 #endif
@@ -5922,7 +5922,7 @@
 #endif //_M_X64
 #endif //Q_OS_WIN
 
-#if defined(Q_OS_LINUX) && !defined(__UCLIBC__)
+#if defined(Q_OS_GLIBC) && !defined(__UCLIBC__)
     fesetenv(&envp);
 #endif
 
